### ISSUE TITLE
Add ability to send copy of SF report to SF Department of Police Accountability 

### ISSUE THREAD

**Issue by BrandonAnderson** [Raheem-ai/raheem-bot#15]
Add ability for user to send copy of their report to SF Department of Police Accountability. 

Might need copy help here.

**Comment by BrandonAnderson** [comment]
![screen shot 2017-04-05 at 4 56 47 pm](https://cloud.githubusercontent.com/assets/12467495/24731757/f7aaa0d0-1a20-11e7-9065-3b67b540a5bf.png)
![screen shot 2017-04-05 at 4 57 02 pm](https://cloud.githubusercontent.com/assets/12467495/24731762/f9bf32dc-1a20-11e7-821a-6ed0b71b5282.png)

**Comment by Jlawlzz** [comment]
@BrandonAnderson @jhankin This could be done via chatbot, but backends are regularly used to do something like this. Thoughts on approach?

**Comment by BrandonAnderson** [comment]
@Jlawlzz: I talked with @jhankin. I think the backend will do it. Can you verify with him, though? I'm concerned about the complexity considering an incident has to have happened in SF for a report to be generated to the SF DPA.

Another thing. Even if the backend does this job, should the chatbot let the user know that the report is going to the SF DPA? Maybe give them the option whether to file submit it there or not?

Example text: "The SF DPA is [value prop]. Would you like submit this report to the SF DPA?"

**Comment by jhankin** [comment]
Giving the reporter the option to send instead of doing it automatically is
a really good call.  I feel like some people might feel violated if we just
submitted the report on their behalf without asking.

It's a little more work but not bad.

On Wed, Apr 5, 2017 at 9:02 PM Brandon Anderson <notifications@github.com>
wrote:

@Jlawlzz <https://github.com/Jlawlzz>: I talked with @jhankin
<https://github.com/jhankin>. I think the backend will do it. Can you
verify with him, though? I'm concerned about the complexity considering an
incident has to have happened in SF for a report to be generated to the SF
DPA.

Another thing. Even if the backend does this job, should the chatbot let
the user know that the report is going to the SF DPA? Maybe give them the
option whether to file submit it there or not?

Example text: "The SF DPA is [value prop]. Would you like submit this
report to the SF DPA?"

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/Raheem-ai/raheem-bot/issues/15#issuecomment-292062984>,
or mute the thread

**Comment by BrandonAnderson** [comment]
@jhankin agreed. How soon can we get the backend drafting summaries and sending them to the DPA? 

- the backend needs the ability to determine whether a user logged an interaction that happened in the City or County of San Francisco. 
- from here, the bot would prompt the user with the question: "would you like me to tell the SF DPA to open an investigation?"

**Comment by jhankin** [comment]
Point 1 is blocked on landing this: https://github.com/Raheem-ai/seed_web/pull/57

From there I guess we will be returning the `Place`s (state, city, county, neighborhood, etc.) associated with the incident every time the incident is updated, so the bot just needs to look at whether one of those `Place`s is San Francisco and ask the question "would you like me to tell the SF DPA to open an investigation?"

We can then either create a new endpoint for the bot to trigger the report, or we can introspect the metadata looking for something like `INFORM_SF_DPA : Yes` as one of the responses.  Each approach has its upside and its downside.  @Jlawlzz @seansu4you87 we should discuss implementation details.

**Comment by jhankin** [comment]
@BrandonAnderson in thinking about this more, I think there's an argument to be made for NOT automating this just yet.  There's a popular and useful notion in the startup world about "doing things that don't scale" in the beginning and worrying about scale when the time is right.  Until we have hundreds or thousands of reports coming in, we don't necessarily have to build a completely automated flow for sending these reports to SF DPA.  

We could conceivably start asking in the chatbot something to the effect of, "Would you mind if I filed an anonymous report with the city on your behalf?  We won't tell them anything personally identifying about you."  Then we could check the reports every week or so and if there are any from SF, we could pull the info and send them an email by hand.  

Just something to think about as we prioritize the work.

**Comment by BrandonAnderson** [comment]
Agreed. We shouldn't build something that we don't currently need. We gotta do two things. 

- add question to the chatbot
- decide on what days we're sending the report to DPA

Jordan and I added this question yesterday: "Can I tell your local jurisdiction to open an investigation about this report? Don't worry, your identity will still be anonymous." 

[Yes] [No]

Though, I sort of like your question better. Thoughts?

**Comment by BrandonAnderson** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Add ability to send copy of SF report to SF Department of Police Accountability #15](https://trello.com/c/qBybPw5C/43-add-ability-to-send-copy-of-sf-report-to-sf-department-of-police-accountability-15)

**Comment by BrandonAnderson** [comment]
@jhankin this is what @Jlawlzz decided on. 

- Will use current copy for the bot: "Can I tell your local jurisdiction to open an investigation about this report?" This avoids us making any promises to the user that WE ourselves can file the report, instead we're just relaying that the user would like an investigation opened.

- Mondays are probably the best day to review data then to forward any SF reports to the DPA. Choosing Monday because people, I presume, have more incidents with police then. Would you like to review every Monday morning and send by evening the reports filed in SF to the DPA?
